libcore/src/events/EventManager.cpp

```diff
@@ -38,74 +38,42 @@ EventManager::EventManager(Building * _b)
     _building = _b;
 }
 
-void EventManager::AddEvents(std::vector<Event> events)
+void EventManager::AddEvent(std::unique_ptr<Event> event)
```

Comment:
Could you use R-value references here to avoid the erroneous use with a unique pointer. With r value ref it is clear, that the memory is claimed in the function.

Reply:
After discussion: no change. From the cpp core guidelines: R.32: Take a unique_ptr<widget> parameter to express that a function assumes ownership of a widget